src/Models/Context/SceneViewContext/SceneViewContext.tsx

```diff
@@ -29,16 +32,92 @@ export const SceneViewContextReducer: (
         logDebugConsole(
             'info',
             `Updating Sceneview context ${action.type} with payload: `,
-            action.payload
+            (action as any).payload // ignore that payload doesn't always come since this is just a log
         );
         switch (action.type) {
-            // Global setting action, overwrites only the values included in the payload
-            case SceneViewContextActionType.SET_SCENE_VIEW_ATTRIBUTES: {
-                Object.keys(action.payload).forEach((key) => {
-                    draft[key] = action.payload[key];
-                });
+            case SceneViewContextActionType.SET_SCENE_OUTLINED_MESHES: {
+                const customMeshItems = createCustomMeshItems(
+                    action.payload.meshIds,
+                    action.payload.color
+                );
+                draft.outlinedMeshItems = customMeshItems;
                 break;
             }
+            case SceneViewContextActionType.RESET_SELECTED_MESHES: {
+                draft.outlinedMeshItems = [];
+                break;
+            }
+            case SceneViewContextActionType.RESET_OUTLINED_MESHES: {
+                draft.outlinedMeshItems = action.payload
+                    ? action.payload.outlinedMeshItems
+                    : [];
+                break;
+            }
+            case SceneViewContextActionType.OUTLINE_BEHAVIOR_MESHES: {
+                const behavior = action.payload.behavior;
+                // get elements that are contained in the hovered behavior
+                let meshIds: string[] = [];
+                const selectedElements: ITwinToObjectMapping[] = [];
+                behavior.datasources
+                    .filter(
+                        ViewerConfigUtility.isElementTwinToObjectMappingDataSource
+                    )
+                    .forEach((ds) => {
+                        ds.elementIDs.forEach((elementId) => {
+                            const element = action.payload.elements.find(
+                                (el) => el.id === elementId
+                            );
+                            element && selectedElements.push(element);
+                        });
+                    });
+
+                for (const element of selectedElements) {
```

Comment:
I haven't looked in detail at what htis actually does, would it make any difference if there are duplicate meshIds? would it make sense to add the items to a set instead of an array?

Reply:
changed this to a set